README.md

```diff
@@ -1,121 +1,123 @@
+* [You can convert from other popular time series toolkits in Python](https://tslearn.readthedocs.io/en/latest/integration_other_software.html)
+* [You can load any of the UCR datasets in the required format.](https://tslearn.readthedocs.io/en/latest/gen_modules/tslearn.datasets.html#module-tslearn.datasets)
+* [You can generate synthetic data using the `generators` module](https://tslearn.readthedocs.io/en/latest/gen_modules/tslearn.generators.html#module-tslearn.generators)
+
+It should further be noted that tslearn [supports variable-length timeseries](https://tslearn.readthedocs.io/en/latest/variablelength.html).
+
+```python3
+>>> from tslearn.utils import to_time_series_dataset
+>>> my_first_time_series = [1, 3, 4, 2]
+>>> my_second_time_series = [1, 2, 4, 2]
+>>> my_third_time_series = [1, 2, 4, 2, 2]
+>>> X = to_time_series_dataset([my_first_time_series,
+                                my_second_time_series,
+                                my_third_time_series])
+>>> y = [0, 1, 1]
+```
 
-# Dependencies
+## 2. Data preprocessing and transformations
+Optionally, tslearn has several utilities to preprocess the data. In order to facilitate the convergence of different algorithms, you can [normalize time series](https://tslearn.readthedocs.io/en/latest/gen_modules/tslearn.preprocessing.html#module-tslearn.preprocessing). Alternatively, in order to speed up training times, one can [resample](https://tslearn.readthedocs.io/en/latest/gen_modules/preprocessing/tslearn.preprocessing.TimeSeriesResampler.html#tslearn.preprocessing.TimeSeriesResampler) the data or apply a [piece-wise transformation](https://tslearn.readthedocs.io/en/latest/gen_modules/tslearn.piecewise.html#module-tslearn.piecewise).
 
+```python3
+>>> from tslearn.preprocessing import TimeSeriesScalerMinMax
+>>> X_scaled = TimeSeriesScalerMinMax().fit_transform(X)
+>>> print(X_scaled)
+[[[0.] [0.667] [1.] [0.333] [nan]]
+ [[0.] [0.333] [1.] [0.333] [nan]]
+ [[0.] [0.333] [1.] [0.333] [0.333]]]
 ```
-Cython
-numpy
-numba
-scipy
-scikit-learn
-joblib
-numba
-```
-
-If you plan to use the `shapelets` module, `keras` and `tensorflow` should also be installed.
-`h5py` is required for reading or writing models using the hdf5 file format.
 
-# Installation
+## 3. Training a model
 
-## Using conda
+After getting the data in the right format, a model can be trained. Depending on the use case, tslearn supports different tasks: classification, clustering and regression. For an extensive overview of possibilities, check out our [gallery of examples](https://tslearn.readthedocs.io/en/latest/auto_examples/index.html).
 
-The easiest way to install `tslearn` is probably via `conda`:
-```bash
-conda install -c conda-forge tslearn
+```python3
+>>> from tslearn.neighbors import KNeighborsTimeSeriesClassifier
+>>> knn = KNeighborsTimeSeriesClassifier(n_neighbors=1)
+>>> knn.fit(X_scaled, y)
+>>> print(knn.predict(X_scaled))
+[0 1 1]
 ```
```

Comment:
Here, I'd promote the fact that `tslearn` models are compatible with `sklearn` and hence cross validation and pipelines can be used (with a link to your gallery example).